### PR TITLE
fix l2arctic dataset prep when output_dir is not None

### PR DESCRIPTION
**Pull request by oplatek**

### CODE REVIEW

File: lhotse/recipes/l2_arctic.py [hunk 1]
```diff
@@ -162,6 +163,8 @@ def prepare_l2_arctic(
     }
 
     if output_dir is not None:
+        output_dir = Path(output_dir)
+        makedirs(output_dir, exist_ok=True)
```
Comment (pzelasko) [inline lhotse/recipes/l2_arctic.py:167]: FYI: Path.mkdir(...)

Comment (csukuangfj) [reply lhotse/recipes/l2_arctic.py:167]: https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir

Comment (csukuangfj) [reply lhotse/recipes/l2_arctic.py:167]: Turns out the two are equivalent.

See

### pathlib.mkdir
https://github.com/python/cpython/blob/2a36b09ce7cd67503893412b53295717d66fee19/Lib/pathlib.py#L1221


```python
    def mkdir(self, mode=0o777, parents=False, exist_ok=False):
```

https://github.com/python/cpython/blob/2a36b09ce7cd67503893412b53295717d66fee19/Lib/pathlib.py#L361

```python
    mkdir = os.mkdir
```

### os.makedirs

https://github.com/python/cpython/blob/2a36b09ce7cd67503893412b53295717d66fee19/Lib/os.py#L200

```python
def makedirs(name, mode=0o777, exist_ok=False):
   ...
    try:
        mkdir(name, mode)
```

Comment (pzelasko) [reply lhotse/recipes/l2_arctic.py:167]: Yeah. My comment was just about consistency (we typically use `pathlib` which is a bit more high-level than  `os`/`os.path` in Lhotse)

**Pull request merged**